src/main/kotlin/org/gradle/script/lang/kotlin/provider/CachingKotlinCompiler.kt

```diff
@@ -94,16 +94,17 @@ class CachingKotlinCompiler(
 
     fun compileBuildScript(
         scriptFile: File,
+        script: String,
         additionalSourceFiles: List<File>,
         classPath: ClassPath,
         parentClassLoader: ClassLoader): CompiledScript {
 
         val scriptFileName = scriptFile.name
-        return compileScript(cacheKeyPrefix + scriptFileName + scriptFile, classPath, parentClassLoader) {
+        return compileScript(cacheKeyPrefix + scriptFileName + script, classPath, parentClassLoader) { cacheDir ->
```

Comment:
This behaves slightly different as `File` and `String` have [different cache key semantics](https://github.com/gradle/gradle/blob/93a68297faf3681679e58253bd73e326f02fa9e4/subprojects/core/src/main/java/org/gradle/cache/internal/DefaultCacheKeyBuilder.java#L63-L63).

I imagine it might have a (very small) performance impact when the same build script file is reused many times in a single build.

Just an observation.